binzx/otomi

```diff
@@ -0,0 +1,307 @@
+
+stack_volume=""
+tmp_volume_dir=$(mktemp -d)
+tmp_volume=" -v $tmp_volume_dir:/tmp"
+if [[ $base_dir == *"otomi-core"* ]] || [[ $(pwd) == *"otomi-core"* ]]; then
+  silent echo "Checking if in dev environment"
+  if [[ $base_dir == *"otomi-core"* ]]; then
+    stack_dir=$(cd "$base_dir/.." && pwd)
+  elif [[ $(pwd) == *"otomi-core"* ]]; then
+    stack_dir=$(pwd | awk -F'otomi-core' '{print $1"otomi-core"}')
+  fi
+
+  stack_volume="-v $stack_dir:$stack_dir"
+  tmp_volume="-v /tmp:/tmp"
+
+  # Use development distribution if exists
+  if [ -f "${stack_dir}/dist/otomi.js" ]; then
+    executable="node --experimental-specifier-resolution=node ${stack_dir}/dist/otomi.js --"
+  else
+    executable="node --no-warnings --experimental-specifier-resolution=node --loader ts-node/esm ${stack_dir}/src/otomi.ts --"
+  fi
+  echo "OTOMI_DEV=true" >>$tmp_env
+fi
+
+check_volume_path() {
+  source=$1
+  dest=$2
+  [ -z $source ] && [ -z $dest ] && echo "" && return 0
+  echo "-v $source:$dest"
+  return 0
+}
+
+cmd="${executable} $*"
+# If command is "otomi bash"
+if [ "$1" = "bash" ] && [ "$#" = "1" ]; then
+  cmd="bash"
+fi
+
+if [ "$OTOMI_TAG" = 'latest' ] || [ "$OTOMI_TAG" = 'master' ] && [ -z $NOPULL ]; then 
+  silent echo "Pulling latest version of the docker image, please wait"
+  silent docker pull $otomi_tools_image
+  status=$?
+  if [ "$status" -ne 0 ]; then
+    echo "Something went wrong when trying to pull '${otomi_tools_image}'"
+    exit $status
+  fi
+fi
+
+it='-it'
+if [[ $calling_args == *'-no-interactive'* ]] ||
+    [[ $calling_args == *'-ni'* ]]; then
+  it=''
+fi
+
+network='--network host'
+if [[ $calling_args == 'server'* ]]; then
+  network="--init -p $OTOMI_PORT:192019"
+fi
+
+mkdir -p /tmp/otomi
+if [ -n "$OTOMI_IN_DOCKER" ]; then
+  $cmd
+  status=$?
+else
+  docker run --rm $it \
+    $linux_workaround \
+    $network \
+    $stack_volume \
+    $tmp_volume \
+    -v /tmp/otomi:/tmp/otomi \
+    $(check_volume_path $HOME/.kube "/home/app/.kube") \
+    $([ -n "$KUBECONFIG" ] && check_volume_path $KUBECONFIG "/home/app/.kube/config") \
+    $(check_volume_path $HOME/.ssh "/home/app/.ssh") \
+    $(check_volume_path $helm_config "/home/app/.config/helm") \
+    $(check_volume_path $HOME/.config/gcloud "/home/app/.config/gcloud") \
+    $(check_volume_path $HOME/.aws "/home/app/.aws") \
+    $(check_volume_path $HOME/.azure "/home/app/.azure") \
+    $(check_volume_path $ENV_DIR "$stack_dir/env") \
+    $(check_volume_path $ENV_DIR "$ENV_DIR") \
+    $(check_volume_path /var/run/docker.sock "/var/run/docker.sock") \
+    --env-file "$tmp_env" \
+    -w "$stack_dir" \
+    "$otomi_tools_image" \
+    bash -c "$cmd"
+  status=$?
+fi
+
+if [[ $calling_args == *'-skip-cleanup'* ]] || [[ $calling_args == *'-s'* ]]; then
+  running_date=$(date -u '+%y%m%d-%H%M')
+  backup_path="/tmp/otomi/otomi-$running_date"
+  mkdir -p "$backup_path"
+  mv "$tmp_env" "$backup_path/running_env"
+else
+  sleep 0.1
```

Comment:
If sleep is used for some reason here please explain it in the comment above that line

Reply:
Sure, here's the reason for documentation reasons: `Docker seems to be slow to release these files, needs to sleep to make sure that the file & volume are released and can be removed`